### ISSUE TITLE
Does it make sense to implement IntoFuture for js_sys::Promise?

### ISSUE THREAD

**Issue by allsey87** [rustwasm/wasm-bindgen#3184]

**Comment by Liamolucko** [comment]
I think it would make sense, but it's a bit technically difficult to pull off. `js-sys` would have to be the one to implement `IntoFuture`, which means we'd need to add a dependency on `wasm-bindgen-futures` to `js-sys`. But, `wasm-bindgen-futures` already depends on `js-sys`, which means we can't do that. We'd have to first change `wasm-bindgen-futures` to use custom bindings instead of `js-sys`, and then it'd be possible for `js-sys` to (probably optionally) depend on `wasm-bindgen-futures` and implement `IntoFuture` for `Promise`.

**Comment by allsey87** [comment]
Right, for some reason I thought the IntoFuture could be defined inside of `wasm-bindgen-futures` but it is actually the `js_sys` crate that defines `Promise`...

**Comment by hamza1311** [comment]
Not to mention, it'll also increase the minimum supported Rust version for js-sys to when `IntoFuture` was stabilized
